tools/cache.py

```diff
@@ -177,8 +177,8 @@ def get(shortname, creator, what=None, force=False, quiet=False, deferred=False)
     logger.info(message)
     utils.safe_ensure_dirs(cachename.parent)
     creator(str(cachename))
-    if not deferred:
-      assert cachename.exists()
+    if not os.getenv('EMBUILDER_PORT_BUILD_DEFERRED'):
+      assert cachename.is_file()
```

Comment:
I would rather have this check done in `tools/ports/__init__.py` if possible and continue to pass 
`deferred` here as an argument.  

The rational being that `EMBUILDER_PORT_BUILD_DEFERRED` should logically only effect ports and I don't really want post-specific stuff here in the general purpose caching code.

Reply:
Yeah, that does make sense; the only problem is that `cache.get()` is called (`via cache.get_lib()`) from every port build script. So we'd have to update all of them to pass through the `deferred` argument manually. Or else come up with some other weird solution (like making a fake/empty archive file that will pass the assertion but then get overwritten when we run ninja or something).

Reply:
Fair enough LGTM.  Maybe add a comment as to why this needs to be here.

Reply:
I can do that (the only other idea I had that didn't involve updating all the port files was creating an empty archive file that would get overwritten when ninja was run. That seemed worse to me than just having the file not be there, but if you'd prefer that let me know).